search-commons/src/main/java/no/unit/nva/search/resource/SimplifiedResourceModelMutator.java

```diff
@@ -0,0 +1,300 @@
+
+        return new OtherIdentifiers(
+                new HashSet<>(scopusIdentifiers),
+                new HashSet<>(cristinIdentifiers),
+                new HashSet<>(handleIdentifiers),
+                new HashSet<>(isbns),
+                new HashSet<>(issns));
+    }
+
+    private RecordMetadata mutateRecordMetadata(JsonNode source) {
+        return new RecordMetadata(
+                source.path(STATUS).textValue(), source.path(CREATED_DATE).textValue(),
+                source.path(MODIFIED_DATE).textValue(), source.path(PUBLISHED_DATE).textValue());
+    }
+
+    private PublishingDetails mutatePublishingDetails(JsonNode source) {
+        return new PublishingDetails(
+                source.path(ENTITY_DESCRIPTION)
+                        .path(REFERENCE)
+                        .path(PUBLICATION_CONTEXT)
+                        .path(ID)
+                        .textValue(),
+                mutateJournalOrPublisher(source),
+                mutateSeries(source),
+                source.path(ENTITY_DESCRIPTION)
+                        .path(REFERENCE)
+                        .path(PUBLICATION_CONTEXT)
+                        .path(NAME)
+                        .textValue(),
+                source.path(ENTITY_DESCRIPTION).path(REFERENCE).path(DOI).textValue());
+    }
+
+    private String mutateJournalOrPublisher(JsonNode source) {
+        return JOURNAL.equals(
+                        source.path(ENTITY_DESCRIPTION)
+                                .path(REFERENCE)
+                                .path(PUBLICATION_CONTEXT)
+                                .path(TYPE)
+                                .textValue())
+                ? JOURNAL
+                : PUBLISHER;
+    }
+
+    private Series mutateSeries(JsonNode source) {
+        var series =
+                source.path(ENTITY_DESCRIPTION)
+                        .path(REFERENCE)
+                        .path(PUBLICATION_CONTEXT)
+                        .path(SERIES);
+
+        if (series.isMissingNode()) {
+            return null;
+        }
+        return new Series(series.path(ID).textValue(), series.path(NAME).textValue());
+    }
+
+    private List<Contributor> mutateContributorsPreview(JsonNode source) {
+        var contributors = new ArrayList();
+        source.path(ENTITY_DESCRIPTION)
+                .path(CONTRIBUTORS_PREVIEW)
+                .iterator()
+                .forEachRemaining(
+                        c -> {
+                            var affiliationNode = c.path(AFFILIATIONS);
+                            var affiliation =
+                                    affiliationNode.isMissingNode()
+                                            ? null
+                                            : new Affiliation(
+                                                    affiliationNode.get(0).path(ID).textValue(),
+                                                    affiliationNode.get(0).path(TYPE).textValue());
```

Comment:
@brinxmat Is this correct? In the full model a Contributor has multiple affiliations. Do we only want 1 or should I keep all? Do we need both id and type?